pkg/volume/csi/csi_plugin.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/golang/glog"
+	api "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+	csipb "k8s.io/kubernetes/pkg/volume/csi/proto/csi"
+)
+
+const (
+	csiName       = "csi"
+	csiPluginName = "kubernetes.io/csi"
+	//csiAddrTemplate = "/var/lib/kubelet/plugins/csi/sockets/%v/kubeletproxy.sock"
+	csiAddrTemplate = "/tmp/kubeletproxy.sock"
+	csiTimeout      = 15 * time.Second
+)
+
+var (
+	// csiVersion supported csi version
+	csiVersion = &csipb.Version{Major: 0, Minor: 1, Patch: 0}
```

Comment:
Start with 0.0.1

Reply:
Dell/EMC CSI mock uses 0.1.0 and it's nice to have some testing driver.

Reply:
@saad-ali I was under the impression `0.1.0` was the target tag for the spec, no ?